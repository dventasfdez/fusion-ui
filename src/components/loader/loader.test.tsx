import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import Loader, { ILoaderProps } from "./loader";

const LoaderExample = (props: ILoaderProps) => <Loader className="stepone-ui" data-testid="loader-test" {...props} />;
const LoaderExampleWithOutTestIdAndClassName = (props: ILoaderProps) => <Loader {...props} />;

it("render Loader and match snap", () => {
  const component = renderer.create(<LoaderExample />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("render Loader spinner and match snap", () => {
  const component = renderer.create(<LoaderExample spinner />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("render Loader spinner with title and subtitle and match snap", () => {
  const component = renderer.create(<LoaderExample spinner title="Title" subtitle="Subtitle" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("render Loader circular and match snap", () => {
  const component = renderer.create(<LoaderExample />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("render Loader oval and match snap", () => {
  const component = renderer.create(<LoaderExample percentage={{ value: 50, show: true }} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("render Loader and match snap without spinners", () => {
  const component = renderer.create(<LoaderExampleWithOutTestIdAndClassName title="title" subtitle="subtitle" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("render Loader spinner and match snap without spinners", () => {
  const component = renderer.create(<LoaderExampleWithOutTestIdAndClassName spinner />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
it("render Loader spinner and match snap without spinners", () => {
  const component = renderer.create(<LoaderExampleWithOutTestIdAndClassName spinner title="Title" subtitle="Subtitle" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("render Loader circular and match snap without spinners", () => {
  const component = renderer.create(<LoaderExampleWithOutTestIdAndClassName />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("render Loader oval and match snap without spinners", () => {
  const component = renderer.create(<LoaderExampleWithOutTestIdAndClassName percentage={{ value: 50, show: true }} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("loader small", () => {
  const { getByTestId } = render(<LoaderExample percentage={{ value: 50, show: false }} />);
  const loader = getByTestId("loader-test-svg-container");
  expect(loader).toHaveClass("svg-container_circular");
});

it("render loader automatically", () => {
  jest.useFakeTimers();
  const { getByTestId } = render(<LoaderExample title="Test Title" subtitle="Test subtitle" automatic percentage={{ show: true }} />);
  const loader = getByTestId("loader-test-svg-container");

  act(() => {
    jest.advanceTimersByTime(6000);
  });

  expect(loader).toHaveAttribute("data-percentage", "100%");
}, 7000);

it("render title and subtitle", () => {
  const { getByTestId } = render(<LoaderExample title="Test Title" subtitle="Test subtitle" />);

  const title = getByTestId("loader-test-title");
  const subtitle = getByTestId("loader-test-subtitle");
  expect(title).toHaveTextContent("Test Title");
  expect(subtitle).toHaveTextContent("Test subtitle");
});

it("render title and subtitle in spinner", () => {
  const { getByTestId } = render(<LoaderExample spinner title="Test Title" subtitle="Test subtitle" />);

  const title = getByTestId("loader-test-title");
  const subtitle = getByTestId("loader-test-subtitle");
  expect(title).toHaveTextContent("Test Title");
  expect(subtitle).toHaveTextContent("Test subtitle");
});

describe("finish loading both cases", () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  it("success case", async () => {
    const { getByTestId } = render(<LoaderExample title="Test Title" subtitle="Test subtitle" success automatic />);
    const loader = getByTestId("loader-test-svg-container");
    act(() => {
      jest.advanceTimersByTime(6000);
    });

    expect(loader).toHaveClass("success");
  }, 10000);

  it("success case with percentages", async () => {
    const { getByTestId } = render(<LoaderExample title="Test Title" subtitle="Test subtitle" success automatic percentage={{ show: true }} />);
    const loader = getByTestId("loader-test-svg-container");
    act(() => {
      jest.advanceTimersByTime(6000);
    });

    expect(loader).toHaveClass("success");
  }, 10000);

  it("error case", async () => {
    const { getByTestId } = render(<LoaderExample title="Test Title" error subtitle="Test subtitle" automatic />);
    const loader = getByTestId("loader-test-svg-container");

    act(() => {
      jest.advanceTimersByTime(6000);
    });

    expect(loader).toHaveClass("error");
  }, 7000);

  it("error case with percentages", async () => {
    const { getByTestId } = render(<LoaderExample title="Test Title" error subtitle="Test subtitle" automatic percentage={{ show: true }} />);
    const loader = getByTestId("loader-test-svg-container");

    act(() => {
      jest.advanceTimersByTime(6000);
    });

    expect(loader).toHaveClass("error");
  }, 7000);
});

it("render spinner", () => {
  const { getByTestId } = render(<LoaderExample spinner />);
  const loader = getByTestId("loader-test-spinner");
  expect(loader).toHaveClass("spinner");
});
