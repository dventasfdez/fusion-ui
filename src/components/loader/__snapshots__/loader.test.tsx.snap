// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Loader and match snap 1`] = `
<div
  className="loader-wrapper tag-ds"
  data-testid="loader-test"
>
  <div
    className="svg-container_circular "
    data-testid="loader-test-svg-container"
  >
    <svg
      className="svg-content"
      fill="none"
      height="32"
      viewBox="0 0 32 32"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="circle-svg"
        d="M16 30C8.26801 30 2 23.732 2 16C2 8.26801 8.26801 2 16 2C23.732 2 30 8.26801 30 16C30 23.732 23.732 30 16 30Z"
        stroke="rgb(0, 154, 204)"
        strokeWidth="4"
      />
    </svg>
  </div>
</div>
`;

exports[`render Loader and match snap without spinners 1`] = `
<div
  className="loader-wrapper "
>
  <div
    className="svg-container_circular "
  >
    <svg
      className="svg-content"
      fill="none"
      height="32"
      viewBox="0 0 32 32"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="circle-svg"
        d="M16 30C8.26801 30 2 23.732 2 16C2 8.26801 8.26801 2 16 2C23.732 2 30 8.26801 30 16C30 23.732 23.732 30 16 30Z"
        stroke="rgb(0, 154, 204)"
        strokeWidth="4"
      />
    </svg>
  </div>
  <div
    className="loader-text-wrapper"
  >
    <span
      className="loader-title"
    >
      title
    </span>
    <span
      className="loader-subtitle"
    >
      subtitle
    </span>
  </div>
</div>
`;

exports[`render Loader circular and match snap 1`] = `
<div
  className="loader-wrapper tag-ds"
  data-testid="loader-test"
>
  <div
    className="svg-container_circular "
    data-testid="loader-test-svg-container"
  >
    <svg
      className="svg-content"
      fill="none"
      height="32"
      viewBox="0 0 32 32"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="circle-svg"
        d="M16 30C8.26801 30 2 23.732 2 16C2 8.26801 8.26801 2 16 2C23.732 2 30 8.26801 30 16C30 23.732 23.732 30 16 30Z"
        stroke="rgb(0, 154, 204)"
        strokeWidth="4"
      />
    </svg>
  </div>
</div>
`;

exports[`render Loader circular and match snap without spinners 1`] = `
<div
  className="loader-wrapper "
>
  <div
    className="svg-container_circular "
  >
    <svg
      className="svg-content"
      fill="none"
      height="32"
      viewBox="0 0 32 32"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="circle-svg"
        d="M16 30C8.26801 30 2 23.732 2 16C2 8.26801 8.26801 2 16 2C23.732 2 30 8.26801 30 16C30 23.732 23.732 30 16 30Z"
        stroke="rgb(0, 154, 204)"
        strokeWidth="4"
      />
    </svg>
  </div>
</div>
`;

exports[`render Loader oval and match snap 1`] = `
<div
  className="loader-wrapper tag-ds"
  data-testid="loader-test"
>
  <div
    className="svg-container "
    data-percentage="50%"
    data-testid="loader-test-svg-container"
  >
    <svg
      className="svg-content"
      fill="none"
      height="32"
      viewBox="0 0 56 32"
      width="56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="oval-svg"
        d="M28 2H40C47.732 2 54 8.26801 54 16C54 23.732 47.732 30 40 30H28H16C8.26801 30 2 23.732 2 16C2 8.26801 8.26801 2 16 2H28Z"
        stroke="rgb(0, 154, 204)"
        strokeLinecap="round"
        strokeWidth="4"
      />
    </svg>
  </div>
</div>
`;

exports[`render Loader oval and match snap without spinners 1`] = `
<div
  className="loader-wrapper "
>
  <div
    className="svg-container "
    data-percentage="50%"
  >
    <svg
      className="svg-content"
      fill="none"
      height="32"
      viewBox="0 0 56 32"
      width="56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="oval-svg"
        d="M28 2H40C47.732 2 54 8.26801 54 16C54 23.732 47.732 30 40 30H28H16C8.26801 30 2 23.732 2 16C2 8.26801 8.26801 2 16 2H28Z"
        stroke="rgb(0, 154, 204)"
        strokeLinecap="round"
        strokeWidth="4"
      />
    </svg>
  </div>
</div>
`;

exports[`render Loader spinner and match snap 1`] = `
<div
  className="spinner tag-ds"
  data-testid="loader-test-spinner"
/>
`;

exports[`render Loader spinner and match snap without spinners 1`] = `
<div
  className="spinner "
/>
`;

exports[`render Loader spinner and match snap without spinners 2`] = `
<div
  className="loader-wrapper "
>
  <div
    className="spinner"
  />
  <div
    className="loader-text-wrapper"
  >
    <span
      className="loader-title"
    >
      Title
    </span>
    <span
      className="loader-subtitle"
    >
      Subtitle
    </span>
  </div>
</div>
`;

exports[`render Loader spinner with title and subtitle and match snap 1`] = `
<div
  className="loader-wrapper tag-ds"
  data-testid="loader-test"
>
  <div
    className="spinner"
    data-testid="loader-test-spinner"
  />
  <div
    className="loader-text-wrapper"
  >
    <span
      className="loader-title"
      data-testid="loader-test-title"
    >
      Title
    </span>
    <span
      className="loader-subtitle"
      data-testid="loader-test-subtitle"
    >
      Subtitle
    </span>
  </div>
</div>
`;
